src/middlewared/middlewared/utils/metrics/meminfo.py

```diff
@@ -0,0 +1,19 @@
+import humanfriendly
+
+
+def get_memory_info() -> dict:
```

Comment:
I don't know what `mapped`, `active`, or `inactive` mean in relation to ZFS ARC so this function needs to be simplified. Furthermore, we can be more efficient here and return a frozen dataclass object so type annotations are a bit more concise and future developers will get nice hints/autocompletion/etc if we have to work on this again.

```
from dataclasses import dataclass

@dataclass(slots=True, frozen=True, kw_only=True)
class MemoryInfo:
    total: int = 0
    """Total memory available in bytes"""


def get_memory_info() -> MemoryInfo:
    total = 0
    with open('/proc/meminfo') as f:
        for line in filter(lambda x: 'MemTotal' in x, f):
            total = int(line.split()[1]) * 1024
    return MemoryInfo(total=total)